referenceframe/model.go

```diff
@@ -321,47 +319,119 @@ func (m *SimpleModel) inputsToFrames(inputs []Input, collectAll bool) ([]*static
 	if len(m.DoF()) != len(inputs) {
 		return nil, NewIncorrectDoFError(len(inputs), len(m.DoF()))
 	}
-	var err error
+
+	// var err error
 	poses := make([]*staticFrame, 0, len(m.OrdTransforms()))
 	// Start at ((1+0i+0j+0k)+(+0+0i+0j+0k)ϵ)
 	composedTransformation := spatialmath.NewZeroPose()
 	posIdx := 0
+
 	// get quaternions from the base outwards.
-	for _, transform := range m.OrdTransforms() {
+	for _, transform := range m.ordTransforms {
 		dof := len(transform.DoF()) + posIdx
 		input := inputs[posIdx:dof]
 		posIdx = dof
 
-		pose, errNew := transform.Transform(input)
+		// fmt.Printf("Lower transform. Name: %v Type: %T\n", transform.Name(), transform)
+		pose, err := transform.Transform(input)
 		// Fail if inputs are incorrect and pose is nil, but allow querying out-of-bounds positions
-		if pose == nil || (err != nil && !strings.Contains(err.Error(), OOBErrString)) {
+		if pose == nil || err != nil {
 			return nil, err
 		}
-		multierr.AppendInto(&err, errNew)
+
 		if collectAll {
 			var geometry spatialmath.Geometry
 			gf, err := transform.Geometries(input)
 			if err != nil {
 				return nil, err
 			}
-			geometries := gf.Geometries()
+
+			geometries := gf.geometries
 			if len(geometries) == 0 {
 				geometry = nil
 			} else {
 				geometry = geometries[0]
 			}
+
 			// TODO(pl): Part of the implementation for GetGeometries will require removing the single geometry restriction
 			fixedFrame, err := NewStaticFrameWithGeometry(transform.Name(), composedTransformation, geometry)
 			if err != nil {
 				return nil, err
 			}
+
 			poses = append(poses, fixedFrame.(*staticFrame))
 		}
+
 		composedTransformation = spatialmath.Compose(composedTransformation, pose)
 	}
+
 	// TODO(rb) as written this will return one too many frames, no need to return zeroth frame
 	poses = append(poses, &staticFrame{&baseFrame{"", []Limit{}}, composedTransformation, nil})
-	return poses, err
+	return poses, nil
+}
+
+// InputsToTransformOpt is like `inputsToFrames` but only returns the end effector pose. That allows
+// us to optimize away the recording of intermediate computations.
+func (m *SimpleModel) InputsToTransformOpt(inputs []Input) (spatialmath.Pose, error) {
```

Comment:
Confirming my understanding: This function is on a frame. It takes the inputs (joint positions for an arm) and returns the pose of the end effector of the arm in reference to the base of the arm. 

Is that correct?

Reply:
correct!